File: tests/gen_coeff.js
function gen_coeff(k, field) {
        let coeffs = [];
        for (let i = 0; i < k-1; i++) {
                let r = 0;
                while (r == 0 || coeffs.includes(r)) {
                        r = Math.floor(Math.random() * field + 1);
                }
                coeffs.push(r);
        }
        return coeffs;
}

let k = 2;
let field = 257;
console.log(gen_coeff(2,257));

File: tests/gen_coeff.py
from secrets import randbelow
def _gen_coeff(k, field):
    ''' Generate coefficients uniformly at random '''
    coeffs = []
    for i in range(k-1):
        r = 0
        while r == 0 or r in coeffs:
            r = randbelow(field)
        coeffs.append(r)
    return coeffs

k = 2
field = 257

print(_gen_coeff(k,field))
